File: Fortran/ex1/batchGradientDescent.f90
    subroutine batchGradientDescent(M, alpha, iterations, X, y, theta)
	implicit none
	
	integer:: i, M, iter, iterations
	real::temp0, temp1, alpha
	real,dimension(2):: theta
	real,dimension(M):: X, y
	
	do iter = 1, iterations    
    	
		temp0 = 0
    	temp1 = 0   
    	do i = 1, M
        	temp0 = temp0 + (theta(1) + theta(2) * X(i) - y(i)) * 1.
        	temp1 = temp1 + (theta(1) + theta(2) * X(i) - y(i)) * X(i);
    	enddo
    
    	   theta(1) = theta(1) - (alpha/m) * temp0
    	   theta(2) = theta(2) - (alpha/m) * temp1
	enddo
	
	end subroutine batchGradientDescent

File: C/ex1/batchGradientDescent.c
int batchGradientDescent(int M, int iterations, double alpha,  double X[], double Y[], double theta[])
{
	int i, iter;
	double temp0, temp1;
	
	for( iter = 0; iter <  iterations; iter++){   
    	    temp0 = 0;
    	    temp1 = 0;
    
    	    for(i=0; i < M; i++){
        	temp0 = temp0 + (theta[0] + theta[1] * X[i] - Y[i]) * 1;
        	temp1 = temp1 + (theta[0] + theta[1] * X[i] - Y[i]) * X[i];
    	    }
    
    	    theta[0] = theta[0] - (alpha/M) * temp0;
    	    theta[1] = theta[1] - (alpha/M) * temp1;
	}
	
	return 0;
}
